src/clib/lib/job_queue/lsf_driver.cpp

```diff
@@ -402,6 +417,117 @@ static void run_bjobs(lsf_driver_type *driver, char *output_file) {
+        end_index--;
+
+    if (end_index >= 0) {
+        int start_index = 0;
+        while (src[start_index] == ' ')
+            start_index++;
+        strip_length = end_index - start_index + 1;
+        target = (char *)calloc(strip_length + 1, sizeof *target);
+        memcpy(target, &src[start_index], strip_length);
+        CHECK_ALLOC(target);
+    } else {
+        /* A blank string */
+        target = (char *)calloc(strip_length + 1, sizeof *target);
+        CHECK_ALLOC(target);
+    }
+
+    target[strip_length] = '\0';
+    return target;
+}
+
+static char *next_line(FILE *stream, bool *at_eof) {
+    long init_pos = ftell(stream);
+    if (init_pos == -1L)
+        throw std::runtime_error(
+            fmt::format("ftell failed: %d/%s \n", errno, strerror(errno)));
+    int end_char;
+    bool dos_newline;
+    int len = 0;
+    bool cont = true;
+
+    {
+        int c;
+        do {
+            c = fgetc(stream);
+            if (c == EOF)
+                cont = false;
+            else {
+                if (c == '\r' || c == '\n')
+                    cont = false;
+                else
+                    len++;
+            }
+        } while (cont);
+        if (c == '\r')
+            dos_newline = true;
+        else
+            dos_newline = false;
+        end_char = c;
+    }
+
+    if (fseek(stream, init_pos, SEEK_SET) != 0)
+        throw std::runtime_error(
+            fmt::format("fseek failed: %d/%s \n", errno, strerror(errno)));
+
+    char *new_line = (char *)calloc(len + 1, sizeof(char));
+    CHECK_ALLOC(new_line);
+    size_t num_read = fread(new_line, sizeof *new_line, len, stream);
+    if (num_read != len)
+        throw std::runtime_error(
+            fmt::format("failed to read line in bjobs output"));
+    new_line[len] = '\0';
+
+    // Skipping the end of line marker(s).
+    fgetc(stream);
+    if (dos_newline)
+        fgetc(stream);
+
+    if (at_eof != NULL) {
+        if (end_char == EOF)
+            *at_eof = true;
+        else
+            *at_eof = false;
+    }
+
+    if (new_line != NULL) {
+        char *strip_line = strip(new_line);
+        free(new_line);
+
+        return strip_line;
+    } else
+        return NULL;
+}
+
+static void skip_line(FILE *stream) {
+    bool cont = true;
+    int line_nr = 0;
+    bool at_eof = false;
+    char c;
+    do {
+        c = fgetc(stream);
+        if (c == EOF)
+            at_eof = true;
+    } while (c != '\r' && c != '\n' && !at_eof);
+
+    // If we have read a \r this is quite probably a DOS formatted
+    // file, and we read another character in the anticipation that it
+    // is a \n character.
+    if (c == '\r') {
+        c = fgetc(stream);
+        if (c != EOF && c != '\n')
+            if (fseek(stream, -1, SEEK_CUR) != 0)
+                throw std::runtime_error(fmt::format("fseek failed: %d/%s \n",
+                                                     errno, strerror(errno)));
```

Comment:
posix docs say "The fseek()  and fseeko()  functions shall return 0 if they succeed. Otherwise, they shall return -1 and set errno to indicate the error."

Reply:
I'll leave it as the original for now. We are removing this after all.